exercise_solutions.tex

```diff
@@ -1467,6 +1467,62 @@ \subsection*{Solution to \cref{ex:is-conn-trunc-functor}}
   &= (t \circ \tprojf k)(a) \tag{since $g$ is a section of $\lam{s} s\circ f$}\\
   & \equiv t(\tproj ka).
 \end{align*}
+\subsection*{Solution to \cref{ex:categorical-connectivity}}
+
+Let us denote by $\mathsf{isConn}(A)$ the connectivity of type $A$. It is known that
+\begin{displaymath}
+ \mathsf{isConn}(A) \simeq \trunc {} A \times \prd{x,y:A} \trunc {} {\id[A]xy}
+\end{displaymath}
+(see \cref{ex:connectivity-inductively}).
+\begin{enumerate}
+ \item Suppose a type $A$ is connected. The goal $\prd{B,C:\type} \isequiv(e_{A,B,C})$ is a mere proposition, so we can assume $a:A$ and $\prd{x,y:A} \trunc {} {x = y}$.
+
+       Let $B,C$ be types. We want to construct $h:(A\to B+C) \to (A\to B)+(A\to C)$ such that $h \circ e = \idfunc$ and $e \circ h = \idfunc$.
+       Let $f:A \to B+C$ be an arbitrary function. We have case analysis on $f(a):B + C$.\\
+       (a) if $f(a) \equiv \inl(u)$ for some $u:B$:
+       We want a function $g:A\to B$ such that $f(x) = \inl(g(x))$ for every $x:A$. Let us define $k:B+C\to B$ by
+       \begin{align*}
+	k(\inl(b)) &\defeq b;\\
+	k(\inr(c)) &\defeq f(a) \enspace,
+       \end{align*}
+       and $g \defeq k \circ f$. Then we have $\inl(g(x)) = \inl(k(f(x))) = f(x)$ for every $x:A$ (by case analysis on $f(x)$ and $\trunc {} {\id[A]{a}{x}}$).
+       With this $g$, we define $h(f) \defeq \inl(g)$. Then we have
```

Comment:
The flow of logic seems wrong here: the function `h` is defined by case analysis on `f(a)`, so for some `f` it will be in case (a) and others it will be in case (b).  Don't we have complete the definition of `h` first, therefore, including both cases (a) and (b), before starting to prove that `h` is an inverse to `e`?